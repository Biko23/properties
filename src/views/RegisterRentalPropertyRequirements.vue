<template>
<div id="main-div">
    <top-nav />
    <main-nav />

    <v-container>
        <div style="text-align: center">
            <h3 style="color: white">Requirements for adding A Rental</h3>
            <h4 style="color: #b9cbdb">Step 1/4</h4>
        </div>
        <br />
        <v-row id="form-row">
            <v-col cols="12" md="12">
                <v-row>
                    <v-col cols="12" md="4">
                        <v-row>
                            <v-col>
                                <v-card class="mx-auto" max-width="400">
                                    <v-img class="white--text align-end" height="200px" src="../assets/requirements.jpeg">
                                        <v-card-title>Requirements</v-card-title>
                                    </v-img>

                                    <v-card-subtitle class="pb-0">
                                        Registering as a Renter
                                    </v-card-subtitle>

                                    <v-card-text class="text--primary">
                                        <div>&#8594; A Renter can be a broker or a Landlord</div>
                                        <div>&#8594;You need to fullfill the following <b style="font-size: 20px; margin-left: 20px;">&#8595;</b></div>
                                    </v-card-text>

                                    <v-card-actions>
                                        <v-btn color="orange lighten-2" @click="showCard1 = !showCard1" text>
                                            Requirements
                                        </v-btn>

                                        <v-spacer></v-spacer>

                                        <v-btn icon @click="showCard1 = !showCard1">
                                            <v-icon>{{ showCard2 ? 'mdi-chevron-up' : 'mdi-chevron-down' }}</v-icon>
                                        </v-btn>
                                    </v-card-actions>

                                    <v-expand-transition>
                                        <div v-show="showCard1">
                                            <v-divider></v-divider>

                                            <v-card-text>
                                                <div>&#8594; Provide more info about yourself <router-link to="profile">here</router-link>.</div><br />
                                                <div>&#8594; Get Approved by SPL team.</div><br />
                                                <div>&#8594; Upon approval you start Listing.</div>
                                            </v-card-text>
                                        </div>
                                    </v-expand-transition>

                                </v-card>
                            </v-col>
                        </v-row>
                    </v-col>
                    <v-col cols="12" md="4">
                        <v-row>
                            <v-col>
                                <v-card class="mx-auto" max-width="400">
                                    <v-img class="white--text align-end" height="200px" src="../assets/sellingsteps.jpg">
                                        <v-card-title>Listing Steps</v-card-title>
                                    </v-img>

                                    <v-card-subtitle class="pb-0">
                                        Steps to register a rental
                                    </v-card-subtitle>

                                    <v-card-text class="text--primary">
                                        <div>&#8594; There are steps to upload a rental</div>
                                        <div>&#8594;These are shown below <b style="font-size: 20px; margin-left: 20px;">&#8595;</b></div>
                                    </v-card-text>

                                    <v-card-actions>
                                        <v-btn color="orange lighten-2" @click="showCard2 = !showCard2" text>
                                            Listing Steps
                                        </v-btn>

                                        <v-spacer></v-spacer>

                                        <v-btn icon @click="showCard2 = !showCard2">
                                            <v-icon>{{ showCard2 ? 'mdi-chevron-up' : 'mdi-chevron-down' }}</v-icon>
                                        </v-btn>
                                    </v-card-actions>

                                    <v-expand-transition>
                                        <div v-show="showCard2">
                                            <v-divider></v-divider>
                                            <v-card-text>
                                                <div>Step 2: Rental info required.</div><br />
                                                <div>Step 3: Neighborhood info required.</div><br />
                                                <div>Step 4: Landmark Info required.</div>
                                            </v-card-text>
                                        </div>
                                    </v-expand-transition>

                                </v-card>
                            </v-col>
                        </v-row>
                    </v-col>
                    <v-col cols="12" md="4">
                        <v-row>
                            <v-col>
                                <v-card class="mx-auto" max-width="400">
                                    <v-img class="white--text align-end" height="200px" src="../assets/approved.jpeg">
                                        <v-card-title>Rental Approval</v-card-title>
                                    </v-img>

                                    <v-card-subtitle class="pb-0">
                                        Steps for Rental approval
                                    </v-card-subtitle>

                                    <v-card-text class="text--primary">
                                        <div>&#8594; There are steps to approve a property</div>
                                        <div>&#8594;These are shown below <b style="font-size: 20px; margin-left: 20px;">&#8595;</b></div>
                                    </v-card-text>

                                    <v-card-actions>
                                        <v-btn color="orange lighten-2" @click="showCard3 = !showCard3" text>
                                            Approval Steps
                                        </v-btn>

                                        <v-spacer></v-spacer>

                                        <v-btn icon @click="showCard3 = !showCard3">
                                            <v-icon>{{ showCard2 ? 'mdi-chevron-up' : 'mdi-chevron-down' }}</v-icon>
                                        </v-btn>
                                    </v-card-actions>

                                    <v-expand-transition>
                                        <div v-show="showCard3">
                                            <v-divider></v-divider>
                                            <v-card-text>
                                                <div>Step 1: On upload rental goes to SPL.</div><br />
                                                <div>Step 2: Due deligence is done.</div><br />
                                                <div>Step 3: Rental gets approved and appear on the website.</div>
                                            </v-card-text>
                                        </div>
                                    </v-expand-transition>

                                </v-card>
                            </v-col>
                        </v-row>
                    </v-col>
                </v-row>
                <v-row>
                    <v-col style="
                display: flex;
                flex-direction: row;
                justify-content: center;
              ">
                        <template v-if="loginState">
                            <router-link to="/register-rental">
                                <v-btn color="primary">Proceed to Next Step</v-btn>
                            </router-link>
                        </template>
                        <template v-else>
                            <p>
                                If it is your first time visiting us, click
                                <router-link to="/signup">
                                    here
                                </router-link>
                                to register or log in 
                                <router-link to="/login">
                                    here
                                </router-link>
                            </p>
                        </template>
                    </v-col>
                </v-row>
            </v-col>
        </v-row>
        <br />
    </v-container>
    <about />
    <Footer />
    <botton-nav />
</div>
</template>

<script>
import TopNav from "@/components/TopNav.vue";
import MainNav from "@/components/MainNav.vue";
import BottonNav from "../components/BottonNav.vue";
import About from "./About.vue";
import Footer from "../components/Footer.vue";
import {
    mapGetters
} from 'vuex';

export default {
    name: "RegisterRentalPropertyRequirement",
    data: () => ({
        showCard1: false,
        showCard2: false,
        showCard3: false
    }),
    components: {
        TopNav,
        MainNav,
        BottonNav,
        About,
        Footer
    },
    methods: {},
    computed: {
        ...mapGetters(["loginState"])
    },
    // currentLoggedinUser
    created() {},
};
</script>

<style scoped>
#main-div {
    background-color: #3b6ef3;
}

#form-row {
    background-color: #e7f0ff;
    width: 1000px;
    margin-left: auto;
    margin-right: auto;
    border-radius: 6px;
}

#description {
    margin-top: -200px;
}

.custom-placeholer-color input::placeholder {
    color: red !important;
    opacity: 1;
}

.custom-label-color .v-label .solo {
    color: blue;
    opacity: 1;
}

#col-hidden {
    display: none;
}

@media only screen and (max-width: 768px) {
    #main-div {
        background-color: #3b6ef3;
    }

    #form-row {
        background-color: #e7f0ff;
        width: auto;
        margin-left: auto;
        margin-right: auto;
        border-radius: 6px;
    }

    #description {
        margin-top: auto;
    }

    #col-hidden {
        display: block;
    }
}

@media only screen and (max-width: 1200px) {
    #main-div {
        background-color: #3b6ef3;
    }

    #form-row {
        background-color: #e7f0ff;
        width: auto;
        margin-left: auto;
        margin-right: auto;
        border-radius: 6px;
    }
}
</style>
