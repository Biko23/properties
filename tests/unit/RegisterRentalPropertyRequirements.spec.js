import { createLocalVue, mount } from '@vue/test-utils'
import RegisterRentalPropertyRequirements from '@/views/RegisterRentalPropertyRequirements.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('RegisterRentalPropertyRequirements.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            postAUserLog: jest.fn()
        }
        getters = {
            loginState: () => ({})
        }
        store = new Vuex.Store({
            actions,
            getters
        })
    })

    const mountFactory = options => {
        return mount(RegisterRentalPropertyRequirements, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

