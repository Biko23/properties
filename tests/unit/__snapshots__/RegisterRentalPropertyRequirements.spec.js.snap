// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterRentalPropertyRequirements.vue should have a match snapshot 1`] = `
<div id="main-div">
  <div class="container">
    <div style="text-align: center;">
      <h3 style="color: white;">Requirements for adding A Rental</h3>
      <h4 style="color: rgb(185, 203, 219);">Step 1/4</h4>
    </div> <br>
    <div id="form-row" class="row">
      <div class="col-md-12 col-12">
        <div class="row">
          <div class="col-md-4 col-12">
            <div class="row">
              <div class="col">
                <div class="mx-auto v-card v-sheet theme--light" style="max-width: 400px;">
                  <div class="v-image v-responsive white--text align-end theme--light" style="height: 200px;">
                    <transition-stub name="fade-transition" mode="in-out">
                      <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                    </transition-stub>
                    <div class="v-responsive__content">
                      <div class="v-card__title">Requirements</div>
                    </div>
                  </div>
                  <div class="v-card__subtitle pb-0">
                    Registering as a Renter
                  </div>
                  <div class="v-card__text text--primary">
                    <div>→ A Renter can be a broker or a Landlord</div>
                    <div>→You need to fullfill the following <b style="font-size: 20px; margin-left: 20px;">↓</b></div>
                  </div>
                  <div class="v-card__actions"><button type="button" class="v-btn v-btn--text theme--light v-size--default orange--text text--lighten-2"><span class="v-btn__content">
                                            Requirements
                                        </span></button>
                    <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></span></button>
                  </div>
                  <transition-stub name="expand-transition">
                    <div style="display: none;">
                      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
                      <div class="v-card__text">
                        <div>→ Provide more info about yourself <a href="#/profile" class="">here</a>.</div><br>
                        <div>→ Get Approved by SPL team.</div><br>
                        <div>→ Upon approval you start Listing.</div>
                      </div>
                    </div>
                  </transition-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="col-md-4 col-12">
            <div class="row">
              <div class="col">
                <div class="mx-auto v-card v-sheet theme--light" style="max-width: 400px;">
                  <div class="v-image v-responsive white--text align-end theme--light" style="height: 200px;">
                    <transition-stub name="fade-transition" mode="in-out">
                      <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                    </transition-stub>
                    <div class="v-responsive__content">
                      <div class="v-card__title">Listing Steps</div>
                    </div>
                  </div>
                  <div class="v-card__subtitle pb-0">
                    Steps to register a rental
                  </div>
                  <div class="v-card__text text--primary">
                    <div>→ There are steps to upload a rental</div>
                    <div>→These are shown below <b style="font-size: 20px; margin-left: 20px;">↓</b></div>
                  </div>
                  <div class="v-card__actions"><button type="button" class="v-btn v-btn--text theme--light v-size--default orange--text text--lighten-2"><span class="v-btn__content">
                                            Listing Steps
                                        </span></button>
                    <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></span></button>
                  </div>
                  <transition-stub name="expand-transition">
                    <div style="display: none;">
                      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
                      <div class="v-card__text">
                        <div>Step 2: Rental info required.</div><br>
                        <div>Step 3: Neighborhood info required.</div><br>
                        <div>Step 4: Landmark Info required.</div>
                      </div>
                    </div>
                  </transition-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="col-md-4 col-12">
            <div class="row">
              <div class="col">
                <div class="mx-auto v-card v-sheet theme--light" style="max-width: 400px;">
                  <div class="v-image v-responsive white--text align-end theme--light" style="height: 200px;">
                    <transition-stub name="fade-transition" mode="in-out">
                      <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                    </transition-stub>
                    <div class="v-responsive__content">
                      <div class="v-card__title">Rental Approval</div>
                    </div>
                  </div>
                  <div class="v-card__subtitle pb-0">
                    Steps for Rental approval
                  </div>
                  <div class="v-card__text text--primary">
                    <div>→ There are steps to approve a property</div>
                    <div>→These are shown below <b style="font-size: 20px; margin-left: 20px;">↓</b></div>
                  </div>
                  <div class="v-card__actions"><button type="button" class="v-btn v-btn--text theme--light v-size--default orange--text text--lighten-2"><span class="v-btn__content">
                                            Approval Steps
                                        </span></button>
                    <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light"></i></span></button>
                  </div>
                  <transition-stub name="expand-transition">
                    <div style="display: none;">
                      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
                      <div class="v-card__text">
                        <div>Step 1: On upload rental goes to SPL.</div><br>
                        <div>Step 2: Due deligence is done.</div><br>
                        <div>Step 3: Rental gets approved and appear on the website.</div>
                      </div>
                    </div>
                  </transition-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col" style="display: flex; flex-direction: row; justify-content: center;"><a href="#/register-rental" class=""><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Proceed to Next Step</span></button></a></div>
        </div>
      </div>
    </div> <br>
  </div>
  <div id="col-hidden">
    <header class="v-sheet theme--dark v-toolbar v-toolbar--dense v-app-bar" style="height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px; background-color: rgb(68, 119, 251); border-color: #4477fb;">
      <div class="v-toolbar__content" style="height: 48px;"><button type="button" value="recent" class="v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default" style="background-color: rgb(68, 119, 251);"><span class="v-btn__content"><img src="https://res.cloudinary.com/diued7ugb/image/upload/v1625949058/sms_black_24dp_3_og1r7y.png" alt="" width="30" srcset=""> <span style="color: rgb(239, 158, 158);">14</span></span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default" style="background-color: rgb(68, 119, 251);"><span class="v-btn__content"><img src="https://res.cloudinary.com/diued7ugb/image/upload/v1625949073/notifications_black_24dp_2_jvvady.png" alt="" width="30" srcset=""> <span style="color: rgb(239, 158, 158);">14</span></span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default" style="background-color: rgb(68, 119, 251);"><span class="v-btn__content"><img src="https://res.cloudinary.com/diued7ugb/image/upload/v1625948894/home_black_24dp_5_sng2rd.png" alt="" width="30" srcset=""></span></button> <button type="button" value="recent" class="v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default" style="width: 80px; margin-left: auto; background-color: rgb(68, 119, 251);"><span class="v-btn__content"><span>Buy/Rent</span></span></button> <button type="button" value="recent" class="v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default" style="width: 60px; margin-left: auto; background-color: rgb(68, 119, 251);"><span class="v-btn__content"><span>Sell</span></span></button></div>
    </header>
  </div>
</div>
`;
